From 2cc907346d8ce73872f7adaedb6100aa5cbe5b65 Mon Sep 17 00:00:00 2001
From: Lukasz Hawrylko <lukasz.hawrylko@intel.com>
Date: Tue, 18 Feb 2020 12:53:33 +0100
Subject: [PATCH] Strip executable file before generating tboot.gz

All TBOOT sections in executable should be shifted to TBOOT_BASE_ADDR, when
building debug target, additional debug symbols are added to executable and
linker does not put them under correct address. One option is to define all
debug sections in linker script, second - strip executable. As debug section is
not required in tboot.gz, second option was chosen.

In Addition to above change, comment section is now moved to proper address.

Signed-off-by: Lukasz Hawrylko <lukasz.hawrylko@intel.com>
---
 .hgignore                | 1 +
 tboot/Makefile           | 5 ++++-
 tboot/common/tboot.lds.x | 1 +
 3 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/.hgignore b/.hgignore
index d2f0b57..ceca9c0 100644
--- a/.hgignore
+++ b/.hgignore
@@ -32,6 +32,7 @@
 ^tboot/tboot-syms$
 ^tboot/tboot.gz$
 ^tboot/tboot.mb2$
+^tboot/tboot.strip$
 ^lcptools/tpmnv_defindex$
 ^lcptools/tpmnv_getcap$
 ^lcptools/tpmnv_lock$
diff --git a/tboot/Makefile b/tboot/Makefile
index 0996e56..fb1df41 100644
--- a/tboot/Makefile
+++ b/tboot/Makefile
@@ -32,7 +32,10 @@ OBJS := $(obj-y)
 
 TARGET_LDS := $(CURDIR)/common/tboot.lds
 
-$(TARGET).gz : $(TARGET)
+$(TARGET).strip : $(TARGET)
+	strip $< -o $@
+
+$(TARGET).gz : $(TARGET).strip
 	gzip -n -f -9 < $< > $@
 
 $(TARGET).mb2 : $(TARGET).strip
diff --git a/tboot/common/tboot.lds.x b/tboot/common/tboot.lds.x
index 2e1e66e..cffdc6f 100644
--- a/tboot/common/tboot.lds.x
+++ b/tboot/common/tboot.lds.x
@@ -32,6 +32,7 @@ SECTIONS
 	*(.text)
 	*(.fixup)
 	*(.gnu.warning)
+	*(.comment)
 	} :text = 0x9090
 
   .rodata : { *(.rodata) *(.rodata.*) }
-- 
2.25.1

